File: example/node/src/provider.ts
import { OpenAIProvider, CircuitBreaker } from "@deepstrike/sdk"

export const breaker = new CircuitBreaker(5, 60_000)

export function makeProvider() {
  const key = process.env.OPENAI_API_KEY ?? ""
  const model = process.env.OPENAI_MODEL ?? "gpt-4o-mini"
  const baseURL = process.env.OPENAI_BASE_URL
  return new OpenAIProvider(key, model, { maxRetries: 2, baseDelay: 500 }, baseURL)
}

File: example/python/meetingmind/provider.py
import os
from deepstrike import OpenAIProvider


def make_provider() -> OpenAIProvider:
    return OpenAIProvider(
        api_key=os.environ["OPENAI_API_KEY"],
        model=os.environ.get("MODEL", "gpt-4o"),
        base_url=os.environ.get("OPENAI_BASE_URL", "https://api.openai.com/v1"),
    )
